core/app/mailers/spree/base_mailer.rb

```diff
@@ -15,7 +15,30 @@ def money(amount, currency = Spree::Config[:currency])
     helper_method :money
 
     def mail(headers = {}, &block)
-      super if Spree::Config[:send_core_emails]
+      super if should_send_mail?
+    end
+
+    private
+
+    def should_send_mail?
+      pref = Spree::Config[:send_core_emails]
+
+      if pref == true || pref == false
+        return pref
+      end
+
+      # let preference be expressed without the Spree:: prefix for convenience
+      class_names = [self.class.name.sub(/\ASpree::/, ''), self.class.name]
+
+      if pref[:only].present? && (pref[:only] & class_names).empty?
```

Comment:
casting the preference value to an Array (  `Array(pref[:only])` ) would allow you also to pass a single class name like:

`only: 'OrderMailer'`